src/gui/FileRevealer.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * FileRevealer.cpp - Helper file for cross platform file revealing
+ *
+ * Copyright (c) 2025 Andrew Wiltshire <aw1lt / at/ proton/ dot/me >
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "FileRevealer.h"
+
+#include <QDesktopServices>
+#include <QDir>
+#include <QProcess>
+#include <QString>
+#include <QUrl>
+#include <optional>
+
+#include "lmmsconfig.h"
+
+namespace lmms {
+bool FileRevealer::s_canSelect = false;
+
+const QString& FileRevealer::getDefaultFileManager()
+{
+	static std::optional<QString> fileManagerCache;
+	if (fileManagerCache.has_value()) { return fileManagerCache.value(); }
+#if defined(LMMS_BUILD_WIN32)
+	fileManagerCache = "explorer";
+#elif defined(LMMS_BUILD_APPLE)
+	fileManagerCache = "open";
+#else
+	QProcess process;
+
+	// Check for XFCE (exo-open)
+	process.start("xdg-current-desktop");
+	process.waitForFinished(1000);
+	QString desktopEnv = QString::fromUtf8(process.readAllStandardOutput()).trimmed().toLower();
+	if (desktopEnv.contains("xfce")) {
+		fileManagerCache = "exo-open";
+		return fileManagerCache.value();
+	}
+
+	if (!desktopEnv.contains("gnome"))
+	{
+		process.start("xdg-mime", {"query", "default", "inode/directory"});
+	} else
+	{
+		process.start("geo", {"mime", "inode/directory"});
+	}
```

Comment:
Sadly, Ubuntu will match on `gnome` via `ubuntu:GNOME` and ubuntu does not ship with `geo`, so this would be a regression over the old behavior.

```suggestion
	// Workaround for xdg-mime on XFCE returning "catfish" instead of "thunar"
	if (desktopEnv.contains("xfce")) {
		fileManagerCache = "exo-open";
		return fileManagerCache.value();
	}
	
	process.start("xdg-mime", {"query", "default", "inode/directory"});
```

Reply:
> Sadly, Ubuntu will match on gnome via ubuntu:GNOME and ubuntu does not ship with geo, so this would be a regression over the old behavior.

Oh... correction, it's a typo... it's `gio` not `geo`.

The output looks like this Ubuntu 20.04, Ubuntu 24.04, Fedora 34, Fedora 40:

```
gio mime inode/directory
```

```
Default application for “inode/directory”: org.gnome.Nautilus.desktop
Registered applications:
	org.gnome.Nautilus.desktop
	org.gnome.baobab.desktop
Recommended applications:
	org.gnome.Nautilus.desktop
	org.gnome.baobab.desktop

```